pkg/server/server.go

```diff
@@ -43,9 +44,10 @@ type Options struct {
 func New(opts Options, cfg *setting.Cfg, httpServer *api.HTTPServer, roleRegistry accesscontrol.RoleRegistry,
 	provisioningService provisioning.ProvisioningService, backgroundServiceProvider registry.BackgroundServiceRegistry,
 	usageStatsProvidersRegistry registry.UsageStatsProvidersRegistry, statsCollectorService *statscollector.Service,
+	secretMigrationServiceProvider secretsMigrations.SecretMigrationServiceProvider,
 ) (*Server, error) {
 	statsCollectorService.RegisterProviders(usageStatsProvidersRegistry.GetServices())
-	s, err := newServer(opts, cfg, httpServer, roleRegistry, provisioningService, backgroundServiceProvider)
+	s, err := newServer(opts, cfg, httpServer, roleRegistry, provisioningService, backgroundServiceProvider, secretMigrationServiceProvider)
 	if err != nil {
```

Comment:
I am wondering if there is a more subtle place for us to add this, for example in the provisioning service before datasources are provisioned, since that is where the conflict was previously. I will defer to some more expert reviewers though.

Reply:
I recommended this approach given we already use it for provisioning and other things.

Reply:
I think having it on the server makes more sense since it also calls the background services from there and this is basically following the same pattern, but instead of running in the background it's blocking.